Hestia/Hestia/frontend/DevicesScreen/UITableViewControllerDevicesMain.cs

```diff
@@ -316,6 +324,61 @@ public void ProcessSpeech(string result)
                     WarningMessage.Display(strings.noDeviceFound, strings.pronounceDeviceNameCorrectly, this);
                 }
             }
+            else if (result.Contains("remove") || result.Contains("delete"))
+            {
+                device = GetDevice(result);
+                if (device != null)
+                {
+                    // Loop over devices until device is found
+                    for (int section = 0; section < ((TableSourceDevicesMain)DevicesTable.Source).serverDevices.Count; section++)
+                    {
+                        var serverDevices = ((TableSourceDevicesMain)DevicesTable.Source).serverDevices[section];
+                        for (int row = 0; row < serverDevices.Count; row++)

```

Comment:
A bit confusing to use serverDevices for both the list and list of lists. Maybe just just serverDevices[section].Count?